src/main/java/org/apache/sling/xss/impl/CustomPolicy.java

```diff
@@ -0,0 +1,265 @@
+                    }
+
+                    if (!styleSeen) {
+                        policyBuilder.allowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                                .matching(cssValidator.newCssAttributePolicy()).onElements(tag.getValue().getName());
+                    }
+                }
+                break;
+            }
+        }
+
+        // disallow style tag on specific elements
+        policyBuilder.disallowAttributes(CssValidator.STYLE_ATTRIBUTE_NAME)
+                .onElements(cssValidator.getDisallowedTagNames().toArray(new String[0]));
+
+        // ---------- dynamic attributes ------------
+        Map<String, Attribute> dynamicAttributes = new HashMap<>();
+        // checks if the dynamic attributes are allowed
+        if (policy.getDirectives().get("allowDynamicAttributes").equals("true")) {
+            dynamicAttributes.putAll(policy.getDynamicAttributes());
+            for (Attribute attribute : dynamicAttributes.values()) {
+                if (attribute.getOnInvalid().equals("removeTag")) {
+                    onInvalidRemoveTagList.add(attribute.getName());
+                }
+
+                List<Pattern> regexsFromAttribute = attribute.getPatternList();
+                if (regexsFromAttribute != null && regexsFromAttribute.size() > 0) {
+                    for (Pattern regex : regexsFromAttribute) {
+                        dynamicAttributesPolicyMap.put(attribute.getName(), newDynamicAttributePolicy(regex));
+                    }
+                }
+                List<String> allowedValuesFromAttribute = attribute.getLiterals();
+
+                if (allowedValuesFromAttribute != null && allowedValuesFromAttribute.size() > 0) {
+                    dynamicAttributesPolicyMap.put(attribute.getName(),
+                            newDynamicAttributePolicy(true, allowedValuesFromAttribute.toArray(new String[0])));
+                }
+
+            }
+        }
+
+        policyFactory = policyBuilder.allowTextIn(CssValidator.STYLE_TAG_NAME).toFactory();
+
+    }
+
+    public PolicyFactory getCustomPolicyFactory() {
+        return policyFactory;
+    }
+
+    public Map<String, AttributePolicy> getDynamicAttributesPolicyMap() {
+        return dynamicAttributesPolicyMap;
+    }
+
+    public List<String> getOnInvalidRemoveTagList() {
+        return onInvalidRemoveTagList;
+    }
+
+    public CssValidator getCssValidator() {
+        return cssValidator;
+    }
+
+    private static Predicate<String> matchesToPatterns(List<Pattern> patternList) {
+        return new Predicate<String>() {
+            @Override
+            public boolean apply(String s) {
+                for (Pattern pattern : patternList) {
+                    if (pattern.matcher(s).matches()) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            public boolean test(String t) {
```

Comment:
This is a bit confusing. You should not override both `test` and `apply`.

Reply:
Resolved: 7d9e82449f514b2fce6ba4c3aa4ec0df70daf8bb